horovod/runner/gloo_run.py

```diff
@@ -74,22 +88,9 @@ def slot_info_to_command(slot_info):
         :param slot_info: host and slot to execute the run command on
         :return:
         """
-        host_name = slot_info.hostname
-        horovod_rendez_env = (
-            'HOROVOD_HOSTNAME={hostname} '
-            'HOROVOD_RANK={rank} '
-            'HOROVOD_SIZE={size} '
-            'HOROVOD_LOCAL_RANK={local_rank} '
-            'HOROVOD_LOCAL_SIZE={local_size} '
-            'HOROVOD_CROSS_RANK={cross_rank} '
-            'HOROVOD_CROSS_SIZE={cross_size} '
-                .format(hostname=host_name,
-                        rank=slot_info.rank,
-                        size=slot_info.size,
-                        local_rank=slot_info.local_rank,
-                        local_size=slot_info.local_size,
-                        cross_rank=slot_info.cross_rank,
-                        cross_size=slot_info.cross_size))
+        env_vars = create_slot_env_vars(slot_info)
+        horovod_rendez_env = " ".join(
```

Comment:
This is a much cleaner implementation, but one side effect is that it has apparently broke a few of our more exceptionally fragile set of tests in `test_spark.py`.  Basically, this version removes a space that is expected here: https://github.com/horovod/horovod/blob/master/test/test_spark.py#L776

If you remove the extra space in the string on that line of the test, those tests should pass.

Reply:
wow.......

Reply:
that's wild, thanks for the pointer!

Reply:
Fixed!